service/src/main/java/bio/terra/workspace/service/workspace/CloudSyncRoleMapping.java

```diff
@@ -110,23 +116,90 @@ public class CloudSyncRoleMapping {
+              "dataproc.jobs.list")
+          .build();
+
+  private final List<String> projectOwnerDataprocPermissions =
+      new ImmutableList.Builder<String>()
+          .addAll(projectReaderDataprocPermissions)
+          .addAll(projectOwnerPermissions)
+          .add(
+              // Dataproc CRUD permissions
+              // TODO: Move read permissions to projectReader and revoke modify permissions once
+              "dataproc.clusters.create",
+              "dataproc.clusters.update",
+              "dataproc.clusters.delete",
+              "dataproc.clusters.use",
+              "dataproc.clusters.start",
+              "dataproc.clusters.stop",
+              "dataproc.jobs.create",
+              "dataproc.jobs.update",
+              "dataproc.jobs.delete",
+              "dataproc.jobs.cancel",
+              "dataproc.autoscalingPolicies.get",
+              "dataproc.autoscalingPolicies.list",
+              "dataproc.autoscalingPolicies.create",
+              "dataproc.autoscalingPolicies.update",
+              "dataproc.autoscalingPolicies.delete",
+              // TODO: Remove all of the following once adding WSM managed dataproc
+              "dataproc.tasks.lease",
+              "dataproc.tasks.listInvalidatedLeases",
+              "dataproc.tasks.reportStatus",
+              "dataproc.agents.get",
+              "dataproc.agents.create",
+              "dataproc.agents.update",
+              "dataproc.agents.delete",
+              "logging.logEntries.create",
+              "monitoring.metricDescriptors.create",
+              "monitoring.metricDescriptors.get",
+              "monitoring.metricDescriptors.list",
+              "monitoring.monitoredResourceDescriptors.get",
+              "monitoring.monitoredResourceDescriptors.list",
+              "monitoring.timeSeries.create",
+              "compute.machineTypes.get",
+              "compute.networks.get",
+              "compute.networks.list",
+              "compute.projects.get",
+              "compute.regions.get",
+              "compute.regions.list",
+              "compute.zones.get",
+              "compute.zones.list")
+          .build();
+
+  @Autowired
+  public CloudSyncRoleMapping(FeatureConfiguration featureConfiguration) {
+    this.featureConfiguration = featureConfiguration;
+  }
+
+  public ImmutableMap<WsmIamRole, CustomGcpIamRole> getCustomGcpProjectIamRoles() {
+    CustomGcpIamRole projectReader =
+        CustomGcpIamRole.of("PROJECT_READER", projectReaderPermissions);
+    CustomGcpIamRole projectWriter =
+        CustomGcpIamRole.of("PROJECT_WRITER", projectWriterPermissions);
+    CustomGcpIamRole projectOwner = CustomGcpIamRole.of("PROJECT_OWNER", projectOwnerPermissions);
+    if (featureConfiguration.isDataprocEnabled()) {
+      projectReader = CustomGcpIamRole.of("PROJECT_READER", projectReaderDataprocPermissions);
+      projectOwner = CustomGcpIamRole.of("PROJECT_OWNER", projectOwnerDataprocPermissions);
```

Comment:
what about writer? i think they should also have data proc permission

Reply:
Because the permissions allow for direct resource creation on gcp, it made sense to limit them to the most priveleged workspace users. Also this isn't the intended long term way of allowing sharing and collaboration for dataproc.

Reply:
but why does reader can access data proc while writer can't.

Reply:
Oh right for read permissions. Let me add read permissions for writers as well.

Reply:
personally for consistency i still think writer should be able to create data proc because writer are able to create other resources in terra workspace.